tls/s2n_kem.c

```diff
@@ -85,12 +138,32 @@ const struct s2n_kem_group s2n_x25519_kyber_512_r3 = {
 const struct s2n_kem_group s2n_x25519_kyber_512_r3 = { 0 };
 #endif
 
+#if EVP_APIS_SUPPORTED && defined(S2N_LIBCRYPTO_SUPPORTS_KYBER)
+const struct s2n_kem_group s2n_x25519_kyber_768_r3 = {
+    .name = "x25519_kyber-768-r3",
+    .iana_id = TLS_PQ_KEM_GROUP_ID_X25519_KYBER_768_R3,
+    .curve = &s2n_ecc_curve_x25519,
+    .kem = &s2n_kyber_768_r3,
+};
+#else
+const struct s2n_kem_group s2n_x25519_kyber_768_r3 = { 0 };
+#endif
```

Comment:
I'm worried about how much conditional compilation this PR includes. The only symbols we need to worry about not existing are in s2n_kyber_evp.c-- can we limit the conditional compilation to that file?

Like maybe s2n_kem_group should have an "available " field like s2n_cipher_suite does. Then this would just be:
```
const struct s2n_kem_group s2n_x25519_kyber_768_r3 = {
    .name = "x25519_kyber-768-r3",
    .iana_id = TLS_PQ_KEM_GROUP_ID_X25519_KYBER_768_R3,
    .curve = &s2n_ecc_curve_x25519,
    .kem = &s2n_kyber_768_r3,
    .available = EVP_APIS_SUPPORTED && S2N_LIBCRYPTO_SUPPORTS_KYBER,
};
```
And you could only choose kem groups that that are available.

Reply:
ah, i like the idea of adding the `available` field to the `s2n_kem_group` struct. I'll see if we set that field at runtime. i'll also try to minimize the conditional compilation outside of `s2n_kyber_evp.c`. I think we'll still need it in `s2n_kem.c` (where we have to [decide whether to use use s2n's Kyber512 implementation or that of the linked libcrypto](https://github.com/aws/s2n-tls/blob/main/tls/s2n_kem.c#L35-L43)), but we should be able to cut the conditional compilation elsewhere.

Reply:
I'm still working on this. there are a lot of assumptions baked into the test code that only viable KEM groups are included in the build, and I'm fighting through a few segfaults across CI compiler dimensions.

Reply:
@lrstewart and I talked offline and agreed to move runtime the KEM detection change to a fourth PR in this series after this PR and the tests in PR #4089 have been merged, separating the concern of adding a new policy from that of refactoring KEM support detection. this work will be tracked in Issue #4100 and PR #4101.